### ISSUE TITLE
Verify-Links.ps1 doesn't fail when GitHub returns 200 "Create New Page"

### ISSUE THREAD

**Issue by hallipr** [Azure/azure-sdk-tools#6196]
If a github wiki link doesn't exist, github returns a "Create New Page" response with a 200 status code.  This should be interpreted as a 404.

There's nothing popping out in the HEAD response headers to differentiate between valid and invalid wiki urls.
The GET response has a predictable "Create New Page" title, but if the HEAD request is successful, we don't do a GET request so we'd never see the title.

We could update Verify-Links.ps1 to do a GET request and a content check for `<title>Create New Page</title>` if the URI is for a github wiki.

**Comment by billwert** [comment]
In doing some experimentation around this space I noticed that the Github wiki will use a 302 redirect when the page isn't there:
```pwsh
➜ iwr -MaximumRedirection 0 https://github.com/Azure/azure-sdk-for-java/wiki/Set-up-Your-Environment-for-Authentication  -SkipHttpErrorCheck

StatusCode        : 302
StatusDescription : Found
Content           :
```

Contrast an existent page:
```pwsh 
➜ iwr -MaximumRedirection 0 https://github.com/Azure/azure-sdk-for-java/wiki/Azure-Identity-Examples#authenticating-in-azure-with-managed-identity

StatusCode        : 200
StatusDescription : OK
Content           :
```

**Comment by hallipr** [comment]
Fixed by #8887
